### PR TITLE
MINOR: Mention that kraft is production ready in upgrade notes

### PR DESCRIPTION
**Pull request by ijuma**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/upgrade.html [hunk 1]
```diff
@@ -81,6 +81,8 @@ <h4><a id="upgrade_3_3_0" href="#upgrade_3_3_0">Upgrading a KRaft-based cluster
 
 <h5><a id="upgrade_330_notable" href="#upgrade_330_notable">Notable changes in 3.3.0</a></h5>
     <ul>
+        <li>KRaft mode is production ready for new clusters. See <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-833%3A+Mark+KRaft+as+Production+Ready">KIP-833</a>
```
Comment (jsancio) [inline docs/upgrade.html:84]: @ijuma should we use this URL https://cwiki.apache.org/confluence/x/8xKhD instead? This URL was generated using confluence share wiki feature.

**Review by jsancio — APPROVED** [review]
LGTM. One minor comment.

File: docs/upgrade.html (hunk 1 above)
Comment (ijuma) [reply docs/upgrade.html:84]: We don't use that for the other KIPs though. Are the main advantages that it's shorter and handles renames better?

Comment (jsancio) [reply docs/upgrade.html:84]: The URL I pasted will never change. The URL with the KIP name can change if the author decides to change the name of the KIP. I forget how Confluence performs the redirect. I think in some cases I have seen Confluence to it automatically in other cases I have seen it suggest a different URL in the rendered webpage.

Comment (jsancio) [reply docs/upgrade.html:84]: > We don't use that for the other KIPs though.

I use that kind of URL when referencing KIPs but maybe others don't.

**Pull request merged**